Formula 1/ViewControllers/CircuitsCollectionViewController.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  TracksViewController.swift
+//  Formula 1
+//
+//  Created by Gio on 11/12/19.
+//  Copyright © 2019 Gio. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import Combine
+import SwiftUI
+
+class CircuitsCollectionViewController: UIViewController {
+    @ObservedObject var viewModel = CircuitsViewModel()
+    
+    var collectionView = CircuitsCollectionView()
+    
+    override func viewDidLoad() {
+        title = "Circuits"
+        
+        addSubviews()
+        activateConstraints()
+        
+        enableDelegates()
+    }
+    
+    func addSubviews() {
+        view.addSubview(collectionView)
+    }
+    
+    func activateConstraints() {
+        collectionView.translatesAutoresizingMaskIntoConstraints = false
+        
+        NSLayoutConstraint.activate([
+            collectionView.topAnchor.constraint(equalTo: view.topAnchor),
+            collectionView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            collectionView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            collectionView.bottomAnchor.constraint(equalTo: view.bottomAnchor)
+        ])
+    }
```

Comment:
I'd combine these two functions into one since adding constraints if the view isn't in the hierarchy would crash. You don't ever want to only call one without the other.

Reply:
I like the separation

Reply:
Let me rephrase, I think this is a bad idea. There's potential here for a crash and an important goal is to **making impossible states impossible**.

Reply:
thank you for clarifying

Reply:
ok, so what's the counter argument?